pkg/provider/server.go

```diff
@@ -0,0 +1,185 @@
+package provider
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"io"
+	"net"
+	"os"
+	"path"
+
+	"sigs.k8s.io/secrets-store-csi-driver/provider/v1alpha1"
+)
+
+type SpringCloudConfigCSIProviderServer struct {
+	grpcServer              *grpc.Server
+	listener                net.Listener
+	socketPath              string
+	returnErr               error
+	errorCode               string
+	springCloudConfigClient SpringCloudConfigClient
+	objects                 []*v1alpha1.ObjectVersion
+	files                   []*v1alpha1.File
+}
+
+type Attributes struct {
+	serverAddress string
+	application   string
+	profile       string
+	fileType      string
+}
+
+func (a *Attributes) verify() error {
+	if a.serverAddress == "" {
+		return fmt.Errorf("serverAddress is not set")
+	}
+
+	if a.application == "" {
+		return fmt.Errorf("application is not set")
+	}
+
+	if a.profile == "" {
+		return fmt.Errorf("profile is not set")
+	}
+
+	if a.fileType == "" {
+		return fmt.Errorf("fileType is not set")
+	}
+
+	return nil
+}
+
+// NewMocKCSIProviderServer returns a mock csi-provider grpc server
+func NewSpringCloudConfigCSIProviderServer(socketPath string) (*SpringCloudConfigCSIProviderServer, error) {
+	client := NewSpringCloudConfigClient()
+	server := grpc.NewServer()
+	s := &SpringCloudConfigCSIProviderServer{
+		springCloudConfigClient: client,
+		grpcServer:              server,
+		socketPath:              socketPath,
+	}
+	v1alpha1.RegisterCSIDriverProviderServer(server, s)
+	return s, nil
+}
+
+// SetReturnError sets expected error
+func (m *SpringCloudConfigCSIProviderServer) SetReturnError(err error) {
+	m.returnErr = err
+}
+
+// SetObjects sets expected objects id and version
+func (m *SpringCloudConfigCSIProviderServer) SetObjects(objects map[string]string) {
+	var ov []*v1alpha1.ObjectVersion
+	for k, v := range objects {
+		ov = append(ov, &v1alpha1.ObjectVersion{Id: k, Version: v})
+	}
+	m.objects = ov
+}
+
+// SetFiles sets provider files to return on Mount
+func (m *SpringCloudConfigCSIProviderServer) SetFiles(files []*v1alpha1.File) {
```

Comment:
I am a bit skeptical about the usefulness of this function. What catches my eye is that it modifies `m.files`. Given that this code runs as a server and requests can come in in any order, this opens the door for requests overriding each others state.
Also, the interface [CSIDriverProviderServer](https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/v0.1.0/provider/v1alpha1/service.pb.go#L756-L763) does not describe this function.
Does code of the provider call this function anywhere?